src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/EventWrapperService.kt

```diff
@@ -0,0 +1,279 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.services
+
+import jakarta.persistence.EntityNotFoundException
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.config.FeatureFlagsService
+import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.gateways.AdjudicationCreationRequest
+import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.gateways.AdjudicationDetailsToPublish
+import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.gateways.LegacyNomisGateway
+import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.gateways.OffenderOicSanctionRequest
+import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.gateways.OicHearingRequest
+import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.gateways.OicHearingResultRequest
+import java.time.Clock
+import java.time.LocalDateTime
+
+@Service
+class EventWrapperService(
+  private val snsService: SnsService,
+  private val clock: Clock,
+  private val identityGenerationService: IdentityGenerationService,
+  private val legacyNomisGateway: LegacyNomisGateway,
+  private val featureFlagsService: FeatureFlagsService,
+  private val auditService: AuditService,
+) {
+
+  fun requestAdjudicationCreationData(prisonerNumber: String): AdjudicationCreationRequest {
+    return if (featureFlagsService.isEmitEventsForAdjudications()) {
+      val bookingId = legacyNomisGateway.getPrisonerInfo(prisonerNumber)?.bookingId ?: throw EntityNotFoundException(prisonerNumber)
```

Comment:
not sure we need the bookingId i think the prisoner number passed in is the same value.  Or is this for some other sort of check for the new feature?

Reply:
I wasn't sure why you captured booking ID and was going to discuss